hw9_oop/HW_OOP.py

```diff
@@ -0,0 +1,455 @@
+        Message: Too slow, too slow! The World is the ultimate Stand. Even without his power to stop time, his speed and power far exceed that of your Star Platinum.
+        Date and time: 2023-02-24 14:35:50.797180
+
+        From: ⭐ ＪｏＪｏ ⭐
+        Message: Ora!Ora!Ora!Ora!
+        Date and time: 2023-02-24 14:35:55.801231
+
+        From: ⭐ ＪｏＪｏ ⭐
+        Message: Yare yare daze...
+        Date and time: 2023-02-24 14:35:40.789306
+
+        From: 🦇 ＤＩＯ 🦇
+        Message: Oh ho!~ Then come as close as you like
+        Date and time: 2023-02-24 14:35:35.784470
+
+        From: ⭐ ＪｏＪｏ ⭐
+        Message: I can't beat the shit out of you without getting closer
+        Date and time: 2023-02-24 14:35:30.753319
+
+        From: 🦇 ＤＩＯ 🦇
+        Message: Oh? You're approaching me? Instead of running away, you're coming right to me?
+        Date and time: 2023-02-24 14:35:25.751524
+battle.get_history_from_time_period(after = datetime(2023, 2, 24, 14, 35, 29),
+                                    before = datetime(2023, 2, 24, 14, 35, 40))
+
+        From: ⭐ ＪｏＪｏ ⭐
+        Message: I can't beat the shit out of you without getting closer
+        Date and time: 2023-02-24 14:36:36.080990
+
+        From: 🦇 ＤＩＯ 🦇
+        Message: Oh ho!~ Then come as close as you like
+        Date and time: 2023-02-24 14:36:31.078827
+​
+Задание 2 (3 балла)
+В питоне как-то слишком типично и неинтересно происходят вызовы функций. Напишите класс Args, который будет хранить в себе аргументы, а функции можно будет вызывать при помощи следующего синтаксиса.
+
+Использовать любые модули нельзя, да и вряд-ли это как-то поможет)
+
+class Args:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+    
+    def __rlshift__(self, function):
+        return function(*self.args, **self.kwargs)
+sum << Args([1, 2])
+3
+(lambda a, b, c: a**2 + b + c) << Args(1, 2, c=50)
+53
+Задание 3 (5 баллов)
+Сделайте класс наследник float. Он должен вести себя как float, но также должен обладать некоторыми особенностями:
+
+При получении атрибутов формата <действие>_<число> мы получаем результат такого действия над нашим числом
+Создавать данные атрибуты в явном виде, очевидно, не стоит
+Подсказка: если в процессе гуглёжки, вы выйдете на такую тему как "Дескрипторы", то это НЕ то, что вам сейчас нужно
+
+Примеры использования ниже
+
+class StrangeFloat(float):
+    def __init__(self, number):
+        self.number = number
+        super().__init__()
+        
+    def __getattr__(self, item):
+        function, number2 = item.split('_')
+        if function == 'add':
+            return StrangeFloat(self.number + float(number2))
+        if function == 'subtract':
+            return StrangeFloat(self.number - float(number2))
+        if function == 'multiply':
+            return StrangeFloat(self.number * float(number2))
+        if function == 'divide':
+            return StrangeFloat(self.number / float(number2))
```

Comment:
Конструктор тут не нужен. Если мы ешо уберём, то будет использован унаследованный от `float` конструктор. В таком случае `self` по сути и будет являться нашим числом, а значит атрибут `number` нам также не нужен
```suggestion
    def __getattr__(self, item):
        function, number2 = item.split('_')
        if function == 'add':
            return StrangeFloat(self + float(number2))
        if function == 'subtract':
            return StrangeFloat(self - float(number2))
        if function == 'multiply':
            return StrangeFloat(self * float(number2))
        if function == 'divide':
            return StrangeFloat(self / float(number2))
```

Reply:
`float(number2)` повторяется 4 раза, можно вынести в отдельную переменную